ui/arduino/store.js

```diff
@@ -1509,7 +1509,7 @@ async function uploadFolder(srcPath, destPath, dataConsumer) {
   let allFiles = await disk.ilistAllFiles(srcPath)
   for (let i in allFiles) {
     const file = allFiles[i]
-    const relativePath = file.path.substring(srcPath.length)
+    const relativePath = file.path.substring(srcPath.length+1)
```

Comment:
this change does not seem to affect the process of uploading recursively. I have tested both and saw no changes in the final outcome.
also here you added 1 and at line 1526 removed the slicing.
not sure what this addresses

Reply:
I think this is obsolete after your merged patch. This would remove a separator from the beginning of a relative path.